### DISCUSSION TITLE
Image Viewer Module bigger-picture

### DISCUSSION THREAD

**Discussion by Gordolin** [hbstack/.github#41] [Q&A] [answered]
The Image - Viewer shows as documented pictures up to its full size.

Unfortunately there is no zoom function in this module as it was implemented in your hbs theme.
Do you plan, to implement this function in the future or is it possible, to change the module?

**Comment by razonyang** [comment]
Hi, I've no plan on it so far, you can zoom with mouse wheel. PR are also welcome, the source code on https://github.com/hbstack/bigger-picture/blob/main/assets/hb/modules/bigger-picture/js/panel.ts.

**Reply by Gordolin** [reply]
you are right, but the zoom works only up to the max resolution of the image. I have unfortunatly some small images with a max size (width or height) less than 500px. So zoom in up to 300% is usefull for me, and of course i do this with mouse or fingers :-).

I'm sorry, I'm not a developer. So i am not able to support your project with development.

You do a great job. I like your theme hbs very much, but your new theme hbstack looks a little bit better to me.
I understand your modular approach, but it is very complex.

**Reply by razonyang** [reply]
So you want to zoom in the image to large than it's size, but I didn't find such option for this, I will ask the upstream library author.

**Comment by razonyang** [comment]
I created a feature request issue hbstack/bigger-picture#129, but no ETA, it's very lower priority, since we already able to zoom in/out with mouse wheel on desktops and two fingers on mobiles, and have better UE than buttons.

**Comment by razonyang** [comment, marked as answer]
Now the `github.com/hbstack/bigger-picture@v0.10.0` allows scaling the images large than their sizes (default to `2`), in your case, you'll need to change the `hb.bigger_picture.scale` parameter as `3`.

```yaml
hb:
  bigger_picture:
    scale: 3
```

See https://hbstack.dev/modules/overview/#upgrade-modules for upgrading module.

**Reply by Gordolin** [reply]
works great, thank you
